src/main/scala/strawman/collection/View.scala

```diff
@@ -167,8 +185,22 @@ object IndexedView {
     def apply(i: Int) = underlying.apply(i)
   }
 
+  class TakeRight[A](underlying: IndexedView[A], n: Int)
+  extends View.TakeRight(underlying, n) with IndexedView[A] {
+    override def iterator() = super.iterator() // needed to avoid "conflicting overrides" error
+    def length = underlying.length min normN
+    def apply(i: Int) = underlying.apply(i)
+  }
```

Comment:
I understand that we want `IndexedViews` to implement an optimized version of `TakeRight` but I don’t get why this implementation would have to extend `View.TakeRight`.